### ISSUE TITLE
2 copies of Snaps/Stories being downloaded now

### ISSUE THREAD

**Issue by Magna-Graecia** [rodit/SnapMod#89]
Previously 1 copy was downloaded to whatever path you had set. I left everything at default as it didn't bother me for the time being. Now 2 copies are being downloaded. One to the path you have set, and another to the downloads folder. Seems to be the same file though because if you delete it from downloads, then it will get deleted from the path as well. However, if you delete it from the path, the other will remain in downloads but say "Unable to open file. 'so and so' file was deleted, moved, or renamed. Tap DOWNLOAD to download the file again, or tap REMOVE to remove it from Downloads." It still takes up the same storage space, so I think a 1 MB snap ends up taking up 2 MB for example, and you can't delete one or the other gets deleted/won't open.

I tried tapping DOWNLOAD just to see what it would do, and the file disappeared from downloads, but I have no idea if it actually downloaded the original file and placed it somewhere else haha. I checked through numerous folders and found nothing.

EDIT: Oh and this only happens when downloading snaps/stories from the in-snap/in-story context menu. When using "save to camera roll" on a snap from within chat, only 1 copy gets downloaded to the Snapchat folder. Basically when files are being downloaded to the SnapMod folder, it downloads 2 copies that are linked to each other or something. One in the SnapMod folder, and one in the downloads folder.

**Comment by Magna-Graecia** [comment]
A solution I'm doing to this right now is copying the file from my SnapMod folder to a different folder, then deleting the one in the Downloads folder, which causes both that one and the one in the SnapMod folder to delete, leaving me with just 1 copy of the file, which is as I want it to be. It's a bit annoying to have to do though of course.

I was able to confirm that the file is indeed occupying double space since it's saved in 2 places, even though they're connected and deleting one will cause the other to get deleted or not open. What I did was I saved a story snap, which saved a copy in my SnapMod folder, and a copy in my Downloads folder. Both files are the exact same size. I then went and deleted the copy in my SnapMod folder, which caused the size of that folder to shrink. I then went and deleted the one in my Downloads folder which is unable to open any longer since it's connected to the one in the SnapMod folder that I just deleted. The size of my Downloads folder would then shrink, indicating that the extra file was indeed occupying storage space in it. So basically every time you're downloading a 1 MB snap as an example, you're going to have twice that amount (2 MB) occupied unless you do my workaround stated above to get left with just 1 file. If you don't do it, then you're effectively cutting your storage space in half :(

**Comment by rodit** [comment]
I am unable to recreate your issue but it looks like I did mess up saving snaps/stories in the latest update (saving saved snaps from chats doesn't work).

SnapMod does absolutely nothing in your downloads folder unless you tell it to.

**Comment by Magna-Graecia** [comment]
Alright so I figured it out (kind of). So it's not actually downloading 2 copies of the file, which is why deleting one affects the other (because it's the same file). Basically the one shown in the Downloads folder is just an icon. In fact, it's not even in the actual 'Download' folder in internal storage. It shows up under the "Downloads" section of categories within the Android file system manager. If I delete the snap from the SnapMod folder, that icon remains in the "Downloads" section. Not sure why, and it's annoying because it's going to clutter things up in that section unless I manually delete them. There isn't actually another copy in the Download folder though. I confirmed this by both going to the actual Download folder using the default Android file explorer/manager (whatever it's actually called), and also using Root Explorer. I also used a file analyzer that shows me how much data is being occupied by different types of files (images, videos, APK's, etc), and the data amount decreased when deleting the file from the SnapMod folder, but then when going to delete the un-openable extra "icon" from within "Downloads" afterward, the data amount didn't increase.

It's weird and confusing, and basically an issue with Android file explorer/manager. On the plus side, deselecting the "Use Android Download Manager" stops this from happening, but then if I do that I will no longer get the "downloading" arrow that shows in the status bar at the top of my device and indicates that saving worked. Not much of an issue either way.
